src/System.Net.Primitives/src/System/Net/IPEndPoint.cs

```diff
@@ -119,6 +119,84 @@ public int Port
             }
         }
 
+        public static bool TryParse(string endPointString, out IPEndPoint result)
+        {
+            return TryParse(endPointString.AsSpan(), out result);
+        }
+
+        public static bool TryParse(ReadOnlySpan<char> endPointSpan, out IPEndPoint result)
+        {
+            result = null;
+            int sliceLength = endPointSpan.Length;  // If there's no port then send the entire string to the address parser
+            int lastColonPos = endPointSpan.LastIndexOf(':');
+
+            // Look to see if this is an IPv6 address with a port.
+            if (lastColonPos > 0 && endPointSpan[lastColonPos - 1] == ']')
+            {
+                sliceLength = lastColonPos;
+            }
+            else if(lastColonPos > 0)
+            {
+                // Look to see if this is IPv4 with a port (IPv6 will have another colon)
+
+                int secondToLastColonPos = -1;
+                
+                // Span does not have an overload for LastIndexOf that takes a starting position; we have no choice but to loop
+                for (int i = lastColonPos - 1; i >= 0; i--)
+                {
+                    if (endPointSpan[i] == ':')
+                    {
+                        secondToLastColonPos = i;
+                        break;
+                    }
+                }
+                // No additional colon = IPv4 with port
+                if (secondToLastColonPos == -1)
+                {
+                    sliceLength = lastColonPos;
+                }
+            }
+
+            if (IPAddress.TryParse(endPointSpan.Slice(0, sliceLength), out IPAddress address))
+            {
+                int port = 0;
+                if (sliceLength < endPointSpan.Length &&
+                    !int.TryParse(endPointSpan.Slice(sliceLength + 1), out port))
+                {
+                    return false;
+                }
+
+                if (port < MinPort || port > MaxPort)
+                {
+                    return false;
+                }
+
+                result = new IPEndPoint(address, port);
+                return true;
```

Comment:
If you instead structured this as:
```C#
if (sliceLength >= endPointSpan.Length ||
    (int.TryParse(endPointSpan.Slice(sliceLength + 1), out port) && port >= MinPort && port <= MaxPort))
{
    result = new IPEndPoint(address, port);
    return true;
}
```
then you can move the `result = null` from the beginning to just before the `return false`, still only have it in the code once, but avoid needing to set `result` twice in the case of success.